ui/app/adapters/ldap/role.js

```diff
@@ -26,12 +29,41 @@ export default class LdapRoleAdapter extends NamedPathAdapter {
     return this.getURL(backend, this.pathForRoleType(type), name);
   }
 
-  query(store, type, query) {
-    const { backend, type: roleType } = query;
-    const url = this.getURL(backend, this.pathForRoleType(roleType));
-    return this.ajax(url, 'GET', { data: { list: true } }).then((resp) => {
-      return resp.data.keys.map((name) => ({ name, backend, type: roleType }));
-    });
+  async query(store, type, query, recordArray, options) {
+    const { partialErrorInfo } = options.adapterOptions || {};
+    const { backend } = query;
+    const roles = [];
+    const errors = [];
+
+    for (const roleType of ['static', 'dynamic']) {
+      const url = this.getURL(backend, this.pathForRoleType(roleType));
+      try {
+        const models = await this.ajax(url, 'GET', { data: { list: true } }).then((resp) => {
+          return resp.data.keys.map((name) => ({ name, backend, type: roleType }));
+        });
+        roles.addObjects(models);
+      } catch (error) {
+        if (error.httpStatus !== 404) {
+          errors.push(error);
+        }
+      }
+    }
+
+    if (errors.length) {
+      const message = `Error fetching roles from ${errors.map((e) => e.path).join(' and ')}`;
+      const errorMessages = errors.map((e) => e.errors).flat();
+      if (errors.length === 2) {
```

Comment:
I ran this approach by design and got the thumbs up. The error message itself states which endpoint(s) failed in a single string and then the errors are displayed individually since there could be multiple from the backend. The `1 error occurred` is actually coming from the backend in this case. I can add the path to each individual error as well though so it's easier to identify.

Reply:
I've updated the screenshot in the description to reflect these changes. Thanks for the suggestion!